pkg/appgw/requestroutingrules.go

```diff
@@ -459,6 +464,43 @@ func (c *appGwConfigBuilder) mergePathMap(existingPathMap *n.ApplicationGatewayU
 	return existingPathMap
 }
 
+func (c *appGwConfigBuilder) getListenerPriorities(cbCtx *ConfigBuilderContext) map[listenerIdentifier]*int32 {
+	prioritySet, priorityNil := false, false
+	priorityExists := make(map[int32]bool)
+	allPriorities := make(map[listenerIdentifier]*int32)
+	for _, ingress := range cbCtx.IngressList {
+		klog.V(5).Infof("Getting Request Routing Rules Priority for Ingress: %s/%s", ingress.Namespace, ingress.Name)
+		azListenerConfigs := c.getListenersFromIngress(ingress, cbCtx.EnvVariables)
+		for listenerID := range azListenerConfigs {
+			if priority, err := annotations.GetRequestRoutingRulePriority(ingress); err == nil {
+				klog.V(5).Infof("Request Routing Rules Priority for Ingress: %s/%s is Priority: %d", ingress.Namespace, ingress.Name, *priority)
+				prioritySet = true
+				if _, value := priorityExists[*priority]; !value {
+					priorityExists[*priority] = true
+				} else {
+					klog.Errorf(
+						"Request Routing Rules Priority for Ingress: %s/%s is duplicated. Priority must be unique across all the request routing rules.",
+						ingress.Namespace,
+						ingress.Name)
+				}
+				allPriorities[listenerID] = priority
```

Comment:
The duplicated priority will still be populated after checking.

Reply:
@3quanfeng indeed but update will fail and the error is logged as we do with the other checks in the method.

What would you suggest?

Reply:
@cmendible , I was thinking to fail fast within AGIC in such case instead of propagating to AppGw update process.

Reply:
wouldn't this be a different behavior from what other methods like `getDefaultFromRule` or `noRulesIngress` do?